File: app/src/main/java/com/starsep/sokoban/gamelogic/Position.java
package com.starsep.sokoban.gamelogic;

public class Position {
	public int y;
	public int x;

	public Position(int y, int x) {
		this.y = y;
		this.x = x;
	}

	public Position() {
		this(0, 0);
	}
}

File: app/src/main/java/com/starsep/sokoban/gamelogic/Position.kt
package com.starsep.sokoban.gamelogic

data class Position @JvmOverloads constructor(val y: Int = 0, val x: Int = 0) {
    operator infix fun plus(other: Position) : Position {
        return Position(y + other.y, x + other.x)
    }
    fun copyOf(): Position {
        return Position(y, x)
    }
}
